### PR TITLE
v6.8

### PR DESCRIPTION
**Pull request by KentoNishi**
## No more unavailable videos!

Previously, YouTube and/or streamers blocked third-party sites from opening some streams. This prevented users from watching membership streams and certain channels in the full-page LiveTL viewer. We've come up with a workaround that forces videos to play anyway! No longer do you need to resort to LiveTL's embedded mode to watch restricted streams.

### CODE REVIEW

**Comment by ChrRubin** [comment]
Chat doesn't seem to work well with VODs that were normally unavailable. Either messages will just fail to load completely or it'll stop showing chunks after finishing the current chunk. Tested with: https://www.youtube.com/watch?v=6WFU2wzPKfA

Also doesn't work for members-only VODs, it shows a slightly different "video is not available" screen:
image

**Comment by KentoNishi** [comment]
@ChrRubin 
image

it works totally fine for me. this uses a new content script so i think u might have to fully reinstall the dev build for it to take effect. can u try that?

**Comment by KentoNishi** [comment]
also i can't test membership streams cuz im a peasant :(

**Comment by ChrRubin** [comment]
Hmm, chat seems to be inconsistent on whether or not its working with this workaround. Every few refreshes of the page the chat will just stop working, even with VODs that are normally available. I'll see if I can find a pattern on what's causing it to break. 

And yeah members-only's won't work, at least from what I've tried with Ina's members-only's.

**Comment by ChrRubin** [comment]
Lol somehow even non-HC chat breaks sometimes, the chat is just empty until I scrub the video.

**Comment by ChrRubin** [comment]
Yeah the workaround seems fine most of the time, and only occasionally chat will break for whatever reason.

If possible I'd like it to use the normal embed by default and only use the workaround for unavailable videos, but that depends on whether the player API exposes something that lets us know that the video is unavailable.

File: src/components/options/Dropdown.svelte [hunk 1]
```diff
@@ -38,6 +38,7 @@
     }
   }
   
+  $: console.log(items);
```
Comment (r2dev2) [inline src/components/options/Dropdown.svelte]: remove

File: src/js/content_scripts/injector.js [hunk 2]
```diff
@@ -188,7 +188,31 @@ async function loaded() {
     window.addEventListener('message', d => {
       iframe.contentWindow.postMessage(d.data, '*');
     });
-  }, undefined, mdiIframeArray);
+  };
+
+  console.log(autoLaunchMode.get());
```
Comment (r2dev2) [inline src/js/content_scripts/injector.js]: remove

File: src/js/utils.js [hunk 3]
```diff
@@ -58,3 +58,12 @@ export const escapeRegExp = text => text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\
 export const getAllVoices = () => window.speechSynthesis?.getVoices() || [];
 export const getAllVoiceNames = () => getAllVoices().map(voice => voice.name);
 export const getVoiceMap = () => new Map(getAllVoices().map(v => [v.name, v]));
+
+function capitalize(s) {
```
Comment (r2dev2) [inline src/js/utils.js]: Can you export this as well?

(edited by Kento, gh is high lmao)

**Review by r2dev2 — CHANGES_REQUESTED** [review]
Just some `console.logs` to remove, we also need to wait for me to finish keyboard shortcuts.

**Comment by ChrRubin** [comment]
Can we get #347 checked and merged for 6.8?

**Comment by KentoNishi** [comment]
> Can we get #347 checked and merged for 6.8? 

Sure yea I'll do it in a bit

**Review by r2dev2 — COMMENTED** [review]
I made the TLTalk ad more concise and apparent and this pr looks good. Waiting for #347 to be merged (I approved that one, @KentoNishi pls review and merge).

**Review by r2dev2 — APPROVED** [review]

**Review by ChrRubin — APPROVED** [review]

**Pull request merged**